pkg/console/controllers/oidcsetup/oidcsetup.go

```diff
@@ -0,0 +1,280 @@
+	authenticationClient configv1client.AuthenticationInterface,
+	consoleOperatorInformer operatorv1informers.ConsoleInformer,
+	crdInformer apiexensionsv1informers.CustomResourceDefinitionInformer,
+	targetNSsecretsInformer corev1informers.SecretInformer,
+	targetNSConfigMapInformer corev1informers.ConfigMapInformer,
+	targetNSDeploymentsInformer appsv1informers.DeploymentInformer,
+	recorder events.Recorder,
+) factory.Controller {
+	c := &oidcSetupController{
+		operatorClient: operatorClient,
+
+		authnLister:               authnInformer.Lister(),
+		consoleOperatorLister:     consoleOperatorInformer.Lister(),
+		crdLister:                 crdInformer.Lister(),
+		targetNSSecretsLister:     targetNSsecretsInformer.Lister(),
+		targetNSDeploymentsLister: targetNSDeploymentsInformer.Lister(),
+		targetNSConfigMapLister:   targetNSConfigMapInformer.Lister(),
+
+		authStatusHandler: status.NewAuthStatusHandler(authenticationClient, api.OpenShiftConsoleName, api.TargetNamespace, api.OpenShiftConsoleOperator),
+	}
+	return factory.New().
+		WithSync(c.sync).
+		ResyncEvery(wait.Jitter(time.Minute, 1.0)).
+		WithFilteredEventsInformers(
+			factory.NamesFilter("authentications.config.openshift.io"),
+			crdInformer.Informer(),
+		).
+		WithInformers(
+			authnInformer.Informer(),
+			consoleOperatorInformer.Informer(),
+			targetNSsecretsInformer.Informer(),
+			targetNSDeploymentsInformer.Informer(),
+			targetNSConfigMapInformer.Informer(),
+		).
+		ToController("OIDCSetupController", recorder.WithComponentSuffix("oidc-setup-controller"))
+}
+
+func (c *oidcSetupController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	statusHandler := status.NewStatusHandler(c.operatorClient)
+
+	if shouldSync, err := c.handleManaged(ctx); err != nil {
+		return err
+	} else if !shouldSync {
+		return nil
+	}
+
+	oidcClientsSchema, err := authnConfigHasOIDCFields(c.crdLister)
+	if err != nil {
+		return statusHandler.FlushAndReturn(err)
+	}
+
+	// the schema is feature-gating this controller, we assume API validation won't
+	// allow authentication/cluster 'Type=OIDC' if the `.status.oidcClients` field
+	// does not exist
+	if !oidcClientsSchema {
+		// reset all conditions set by this controller
+		statusHandler.AddConditions(status.HandleProgressingOrDegraded("OIDCClientConfig", "", nil))
+		statusHandler.AddConditions(status.HandleProgressingOrDegraded("AuthStatusHandler", "", nil))
+		return statusHandler.FlushAndReturn(nil)
+	}
+
+	operatorConfig, err := c.consoleOperatorLister.Get(api.ConfigResourceName)
+	if err != nil {
+		return err
+	}
+
+	authnConfig, err := c.authnLister.Get(api.ConfigResourceName)
+	if err != nil {
+		return err
+	}
+
+	// we need to keep track of errors during the sync so that we can requeue
+	// if any occur
+	var errs []error
+	if authnConfig.Spec.Type == configv1.AuthenticationTypeOIDC {
+		syncErr := c.syncAuthTypeOIDC(ctx, syncCtx, statusHandler, operatorConfig, authnConfig)
```

Comment:
without the nesting, this entire function can be inlined to avoid the authSTatus side effect.

And once that is done, authStatus becomes a per-sync function variable, not an instance variable which better represents the state and prevents accidents.